staging/src/k8s.io/code-generator/cmd/informer-gen/generators/factory.go

```diff
@@ -149,8 +162,9 @@ func (f *sharedInformerFactory) WaitForCacheSync(stopCh <-chan struct{}) map[ref
        return res
 }
 
-// InternalInformerFor returns the SharedIndexInformer for obj using an internal
-// client.
+// InformerFor returns the SharedIndexInformer for obj. If the factory has already been started and
+// this is the first time this particular informer has been referenced, the informer will be
+// automatically started. Otherwise, the informer will be started when Start() is invoked.
```

Comment:
Does this sort of thing need codified at the interface level?

Reply:
Probably not